src/Festung/Vault/VaultHandler.hs

```diff
@@ -109,6 +113,26 @@ query o v q = flattenError <$> sendMappedCommand f v q
 
 
 -- | Execute a parametrized query
-parametrizedQuery :: VaultOpener -> Vault -> String -> [P.Value] -> IO (Either VaultError P.QueryResult)
-parametrizedQuery o v q p = flattenError <$> sendMappedCommand f v (q, p)
+parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams
+                  -> IO (Either VaultError P.QueryResult)
+parametrizedQuery o v q (P.Params p) = flattenError <$> sendMappedCommand f v (q, p)
     where f = (,) o . ParametrizedQuery
+parametrizedQuery o v q (P.BulkParams bps) = 
+    _parametrizedQuery o v q (P.BulkParams bps) (Right P.emptyQueryResult)
+
+
+_parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams
+                   -> Either VaultError P.QueryResult -> IO (Either VaultError P.QueryResult)
+_parametrizedQuery o v q (P.BulkParams []      ) res    = return res
+_parametrizedQuery o v q (P.BulkParams (p:bp)) prevRes  = do
+    res <- parametrizedQuery o v q (P.Params p)
+    case res of
+        Left  _  -> return res
+        Right qr -> _parametrizedQuery o v q (P.BulkParams bp) (accumQueryResult prevRes res)
+
+
+accumQueryResult :: Either VaultError P.QueryResult -> Either VaultError P.QueryResult
```

Comment:
I would suggest to create a type alias (maybe `Result` but you'll probably find something better), like so
```haskell
type Result = Either VaultError P.QueryResult
```
so that you can write the type signature as
```haskell
accumulateQueryResult :: Result -> Result -> Result
```
and also the signature of `_parametrizedQuery` would read like
```haskell
_parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams -> Result -> IO (Result)
```
which is still not optimal but reads a little better. As I said `Result` might not be the optimal name, but I can't think of anything more suitable right now.

As we're speaking of type aliases. I think it would be good to have an alias like
```haskell
type Query = String
```
so that the signature of `paramatrizedQuery` can be written as
```haskell
parametrizedQuery :: VaultOpener -> Vault -> Query -> P.QueryParams -> IO (Result)
```
**Edit**: Maybe one could "overload" the `QueryResult` type instead of using the name `Result` as in
```haskell
type QueryResult = Either VaultError P.QueryResult
```